tests/src/test/scala/cats/tests/AlgebraInvariantTests.scala

```diff
@@ -22,21 +22,157 @@ class AlgebraInvariantTests extends CatsSuite {
   val genMonoidInt: Gen[Monoid[Int]] =
     Gen.oneOf(implicitly[Monoid[Int]], intMultiplication, maxInt)
 
+  val genCommutativeMonoidInt: Gen[CommutativeMonoid[Int]] =
+    Gen.oneOf(implicitly[CommutativeMonoid[Int]], intMultiplication)
+
+  val genBoundedSemilatticeSetInt: Gen[BoundedSemilattice[Set[Int]]] =
+    Gen.const(implicitly[BoundedSemilattice[Set[Int]]])
+
+  val genCommutativeGroupInt: Gen[CommutativeGroup[Int]] =
+    Gen.const(implicitly[CommutativeGroup[Int]])
+
   implicit val arbMonoidInt: Arbitrary[Monoid[Int]] =
     Arbitrary(genMonoidInt)
 
-  implicit val arbSemigoupInt: Arbitrary[Semigroup[Int]] =
+  implicit val arbSemigroupInt: Arbitrary[Semigroup[Int]] =
     Arbitrary(genMonoidInt)
 
+  implicit val arbCommutativeMonoidInt: Arbitrary[CommutativeMonoid[Int]] =
+    Arbitrary(genCommutativeMonoidInt)
+
+  implicit val arbCommutativeSemigroupInt: Arbitrary[CommutativeSemigroup[Int]] =
+    Arbitrary(genCommutativeMonoidInt)
+
+  implicit val arbBandSetInt: Arbitrary[Band[Set[Int]]] =
+    Arbitrary(genBoundedSemilatticeSetInt)
+
+  implicit val arbSemilatticeSetInt: Arbitrary[Semilattice[Set[Int]]] =
+    Arbitrary(genBoundedSemilatticeSetInt)
+
+  implicit val arbBoundedSemilatticeSetInt: Arbitrary[BoundedSemilattice[Set[Int]]] =
+    Arbitrary(genBoundedSemilatticeSetInt)
+
+  implicit val arbGroupInt: Arbitrary[Group[Int]] =
+    Arbitrary(genCommutativeGroupInt)
+
+  implicit val arbCommutativeGroupInt: Arbitrary[CommutativeGroup[Int]] =
+    Arbitrary(genCommutativeGroupInt)
+
+  test("Semigroup combineAllOption after imap with id is consistent"){
```

Comment:
I propose we add this to the Semigroup law as a "Ref" law like these ones 
https://github.com/typelevel/cats/blob/master/laws/src/main/scala/cats/laws/FoldableLaws.scala#L122-L143

Reply:
Ah, I didn't know we had such a convention, good to know :)

Reply:
We dont' really, but I'd like to establish one, see discussion [here](https://github.com/typelevel/cats/issues/1908#issuecomment-328899290). @peterneyens has some interesting ideas too.